integration/linker/link-packages-test.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Test that collects all partially built NPM packages and links their Angular
+ * declarations to the corresponding definitions.
+ */
+
+const {NodeJSFileSystem} = require('@angular/compiler-cli/src/ngtsc/file_system');
+const {ConsoleLogger, LogLevel} = require('@angular/compiler-cli/src/ngtsc/logging');
+const {createEs2015LinkerPlugin} = require('@angular/compiler-cli/linker/babel');
+const {getNpmPackagesFromRunfiles} = require('../runfile-helpers');
+const {readFileSync} = require('fs');
+const {join} = require('path');
+const babel = require('@babel/core');
+const {default: traverse} = require('@babel/traverse');
+const glob = require('glob');
+const chalk = require('chalk');
+
+/** File system used by the Angular linker plugin. */
+const fileSystem = new NodeJSFileSystem();
+/** Logger used by the Angular linker plugin. */
+const logger = new ConsoleLogger(LogLevel.info);
+/** List of NPM packages available in the Bazel runfiles. */
+const npmPackages = getNpmPackagesFromRunfiles();
+/** Whether any package could not be linked successfully. */
+let failedPackages = false;
+
+// Iterate through all determined NPM packages and ensure that entry point
+// files can be processed successfully by the Angular linker.
+for (const pkg of npmPackages) {
+  const {failures, passedFiles} = testPackage(pkg);
+
+  console.info(chalk.cyan(`------- Package: @angular/${pkg.name} -------`));
+  console.info(`Passed files: ${passedFiles.length}`);
+  console.info(`Failed files: ${failures.length}`);
+
+  if (failures.length > 0) {
+    failures.forEach(({debugFileName, error}) => {
+      console.error(` •  ${chalk.yellow(debugFileName)}: ${error}`);
+    });
+    failedPackages = true;
+  }
+
+  console.info('-------------------------------------');
+  console.info();
+}
+
+if (failedPackages) {
+  console.error(chalk.red(`✘ Not all packages could be linked successfully. See errors above.`));
+  // If there are failures, exit the process with a non-zero exit code. Bazel
+  // uses exit code `3` to indicate non-fatal test failures.
+  process.exitCode = 3;
+} else {
+  console.info(chalk.green('✓ All packages have been successfully linked.'));
+}
+
+/**
+ * Tests the specified package against the Angular linker plugin.
+ * @param pkg Package being tested.
+ * @returns An object containing linker failures and passed files.
+ */
+function testPackage(pkg) {
+  const entryPointFesmFiles = glob.sync(`fesm2015/**/*.js`, {cwd: pkg.pkgPath});
+  const passedFiles = [];
+  const failures = [];
+
+  // Iterate through each entry point and confirm it's found Angular declarations from a partial
+  // compilation can be linked to the corresponding Angular definitions without errors.
```

Comment:
Actually, I think this whole sentence is a bit confusing. Could we go with:

```suggestion
  // Iterate through each entry point and confirm that all partial declarations can be linked
  // to their corresponding Angular definitions without errors.
```